panic: runtime error: invalid memory address or nil pointer dereference

<!-- 
1. Please search to see if an issue already exists for the bug you encountered.
2. For support requests, FAQs or "How to" questions, please use the GitHub Discussions section instead - https://github.com/projectdiscovery/tlsx/discussions or
3. Join our discord server at https://discord.gg/projectdiscovery and post the question on the #tlsx channel.
-->

<!-- ISSUES MISSING IMPORTANT INFORMATION MAY BE CLOSED WITHOUT INVESTIGATION. -->

### tlsx version:
dev | main

### Current Behavior:

```console
echo 34.107.18.135 | ./tlsx
  

  _____ _    _____  __
 |_   _| |  / __\ \/ /
   | | | |__\__ \>  < 
   |_| |____|___/_/\_\	v1.0.6

		projectdiscovery.io

[INF] Current tlsx version v1.0.6 (latest)
[INF] Connections made using crypto/tls: 0, zcrypto/tls: 0, openssl: 0
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x100a6e1fe]

goroutine 83 [running]:
github.com/projectdiscovery/tlsx/pkg/tlsx/openssl.getResponse({0x100ed44a0, 0xc00049f050}, 0xc000572ae0)
	/Users/geekboy/Github/tlsx/pkg/tlsx/openssl/openssl_exec.go:101 +0x59e
github.com/projectdiscovery/tlsx/pkg/tlsx/openssl.(*Client).ConnectWithOptions(0xc0000364b0, {0xc0001e79c0, 0xd}, {0x0?, 0x0?}, {0x100c18799, 0x3}, {{0x0, 0x0}, {0x0, ...}, ...})
	/Users/geekboy/Github/tlsx/pkg/tlsx/openssl/openssl.go:72 +0x3f5
github.com/projectdiscovery/tlsx/pkg/tlsx/auto.(*Client).ConnectWithOptions(0xc00038c048, {0xc0001e79c0, 0xd}, {0x0, 0x0}, {0x100c18799, 0x3}, {{0x0, 0x0}, {0x0, ...}, ...})
	/Users/geekboy/Github/tlsx/pkg/tlsx/auto/auto.go:56 +0x245
github.com/projectdiscovery/tlsx/pkg/tlsx.(*Service).ConnectWithOptions(0xc00038c000, {0xc0001e79c0, 0xd}, {0x0, 0x0}, {0x100c18799, 0x3}, {{0x0, 0x0}, {0x0, ...}, ...})
	/Users/geekboy/Github/tlsx/pkg/tlsx/tlsx.go:74 +0x243
github.com/projectdiscovery/tlsx/internal/runner.(*Runner).processInputElementWorker(0xc00031b080, 0x0?, 0x0?)
	/Users/geekboy/Github/tlsx/internal/runner/runner.go:179 +0x418
created by github.com/projectdiscovery/tlsx/internal/runner.(*Runner).Execute
	/Users/geekboy/Github/tlsx/internal/runner/runner.go:140 +0x65
```

### Expected Behavior:
parsing cert without crash

### Steps To Reproduce:
```
echo 34.107.18.135 | tlsx
```

### Anything else:
https://github.com/projectdiscovery/httpx/issues/1079

--- a/pkg/tlsx/openssl/openssl_exec.go
+++ b/pkg/tlsx/openssl/openssl_exec.go
@@ -93,17 +93,16 @@ func getResponse(ctx context.Context, opts *Options) (*Response, errorutils.Erro
 	response.Session, errParseSessionData = readSessionData(result.Stdout)
 
 	var allerrors errorutils.Error
-	switch {
-	case errParseCertificates != nil:
+	if errParseCertificates != nil {
 		allerrors = Wrap(allerrors, errorutils.NewWithErr(errParseCertificates).WithTag(PkgTag).Msgf("failed to parse server certificate from response"))
-		fallthrough
-	case errParseSessionData != nil:
+	}
+	if errParseSessionData != nil {
 		allerrors = Wrap(allerrors, errorutils.NewWithErr(errParseSessionData).WithTag(PkgTag).Msgf("failed to parse session data from response"))
-		fallthrough
-	case !opts.SkipCertParse && len(response.AllCerts) == 0:
+	}
+	if !opts.SkipCertParse && len(response.AllCerts) == 0 {
 		allerrors = Wrap(allerrors, errorutils.NewWithTag(PkgTag, "no server certificates found"))
-		fallthrough
-	case allerrors != nil:
+	}
+	if allerrors != nil {
 		// if any of above case is successful
 		// add openssl response
 		return nil, allerrors.Msgf("failed to parse openssl response. original response is:\n%v", *result).Msgf(commadFormat, result.Command)

--- a/pkg/tlsx/ztls/ztls.go
+++ b/pkg/tlsx/ztls/ztls.go
@@ -153,7 +153,9 @@ func (c *Client) ConnectWithOptions(hostname, ip, port string, options clients.C
 		CertificateResponse: ConvertCertificateToResponse(c.options, hostname, ParseSimpleTLSCertificate(hl.ServerCertificates.Certificate)),
 		ServerName:          config.ServerName,
 	}
-	response.Untrusted = clients.IsZTLSUntrustedCA(hl.ServerCertificates.Chain)
+	if response.CertificateResponse != nil {
+		response.Untrusted = clients.IsZTLSUntrustedCA(hl.ServerCertificates.Chain)
+	}
 
 	if c.options.TLSChain {
 		for _, cert := range hl.ServerCertificates.Chain {

